### PR TITLE
Added blank option to the dropdown field

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
If the dropdown field has a blank option, then it should be rendered.

### CODE REVIEW

**Comment by nialldonnellyfh** [comment]
Changes applied to Populated to [this client app](http://testing.ndonnelly.skunkhenry.com/#projects/7ma5peezyzkucryhfgismurz/apps/7ma5pehu6ncjgl2al2rzbxem/details)

This is the [existing client app](http://testing.ndonnelly.skunkhenry.com/#projects/7ma5peezyzkucryhfgismurz/apps/7ma5peh4je6vgvb5s5gj7fjr/details)

File: src/appforms/tests/modules.js [hunk 1]
```diff
@@ -43,7 +44,35 @@
   "060-RuleEngine"
  ],
  backbone:[
-  "040-view06Steps"
```
Comment (danbev) [inline src/appforms/tests/modules.js]: I'm sure I'm missing something here but some of the test listed here do not exist as far as I can tell?

Comment (nialldonnellyfh) [inline src/appforms/tests/modules.js]: That is true, but to be able to test some fields, it has to load the view file also.

If the test file doesn't exist, it won't fail any tests.

Comment (danbev) [inline src/appforms/tests/modules.js]: Ah good to know, thanks!

File: src/appforms/tests/tests/core/040-Model07FieldSelect.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    //The first label should be an empty string if the blank option is included
+    assert.equal("", dropdownOptions[0].label);
+  });
+
+  it("Get Options, blank option included", function() {
+    var dropdownField = new appForm.models.Field(getMockDrowdownFieldData(false));
+
+    var dropdownOptions = dropdownField.getDropdownOptions();
+    assert.equal(2, dropdownOptions.length, "Expected 2 dropdown options");
+    //The first label should be an empty string if the blank option is included
```
Comment (danbev) [inline src/appforms/tests/tests/core/040-Model07FieldSelect.js]: Perhaps this comment should be removed as it seems more related to the previous test?

File: src/appforms/tests/tests/core/040-Model07FieldSelect.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      "repeating": false
+    };
+  }
+
+  it("Get Options, no blank option included", function() {
+    var dropdownField = new appForm.models.Field(getMockDrowdownFieldData(true));
+
+    var dropdownOptions = dropdownField.getDropdownOptions();
+    assert.equal(3, dropdownOptions.length, "Expected 3 dropdown options");
+    //The first label should be an empty string if the blank option is included
```
Comment (danbev) [inline src/appforms/tests/tests/core/040-Model07FieldSelect.js]: Perhaps this comment could be added to the error message if the assert fails:
```javascript
assert.equal("", dropdownOptions[0].label, "First label should be an empty string if the blank option is included");
```

**Comment by danbev** [comment]
👍  Minor optional comments/suggestions above. Looks good to me!

File: src/appforms/tests/tests/core/040-Model07FieldSelect.js (hunk 2 above)
Comment (nialldonnellyfh) [inline src/appforms/tests/tests/core/040-Model07FieldSelect.js]: Yes, typo on my part. Will change, thanks.

File: src/appforms/tests/tests/core/040-Model07FieldSelect.js (hunk 3 above)
Comment (nialldonnellyfh) [inline src/appforms/tests/tests/core/040-Model07FieldSelect.js]: Yep, good idea. I will add that.

**Comment by nialldonnellyfh** [comment]
retest this please

**Comment by nialldonnellyfh** [comment]
retest this please

**Pull request merged**